[Display_01] Create Public Interfaces and Shared Data Types

reedsikorskiOU opened the pull request:
Closes #26 

Created DisplayGpsInterface.java and DisplayGpsInterfaceImplementation.java preGradle Testing

reedsikorskiOU committed:
[Display_01] Created DisplayGpsInterface.java and DisplayGpsInterfaceImplementation.java

Created DisplayGpsInterface.java and DisplayGpsInterfaceImplementation.java preGradle Testing

John-A11 committed:
[Display_01] Created SatelliteSignalCheckRequest

John-A11 committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

reedsikorskiOU committed:
[Display_01] Updated Files to Pass Google Style Guide

reedsikorskiOU committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

John-A11 committed:
Delete SatelliteSignalCheckRequest.java~

reedsikorskiOU committed:
Final Google Style Guide Verification

reedsikorskiOU committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

reedsikorskiOU committed:
Added Import to Classes

John-A11 committed:
Merge branch 'main' into display01-public-interfaces

reedsikorskiOU committed:
Fixing Gradle Error

reedsikorskiOU committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

reedsikorskiOU committed:
plz wrk

John-A11 committed:
Update SatelliteSignalCheckRequest.java

Adjusted spacing according to Google Java Style Guide.

John-A11 committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

John-A11 committed:
Update SatelliteSignalCheckRequest.java

Adjusted spacing according to Google Java Style Guide.

atpoynter committed:
[Display_01] Created DisplayGpsManagerInterface.java and DisplayGpsManagerImplementation.java

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java:
```diff
@@ -0,0 +1,15 @@
+package edu.oakland.production;
```
```suggestion
package edu.oakland.production.display01;
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java:
```diff
@@ -0,0 +1,23 @@
+package edu.oakland.production;
```
```suggestion
package edu.oakland.production.display01;
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsManager.java:
```diff
@@ -0,0 +1,12 @@
+package edu.oakland.production;
```
```suggestion
package edu.oakland.production.display01;
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java:
```diff
@@ -0,0 +1,17 @@
+package edu.oakland.production;
```
```suggestion
package edu.oakland.production.display01;
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java:
```diff
@@ -0,0 +1,17 @@
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+
+public class DisplayGpsManagerImplementation implements DisplayGpsManager{
```
```suggestion
public class DisplayGpsManagerImplementation implements DisplayGpsManager {
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java:
```diff
@@ -0,0 +1,15 @@
+
+public interface DisplayGpsInterface {
+
+  public void recieveGpsSingnals(Satellite satelliteSignal);
```
```suggestion
  public void receiveGpsSignals(Satellite satelliteSignal);
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java:
```diff
@@ -0,0 +1,23 @@
+
+public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+
+  public void recieveGpsSingnals(Satellite satelliteSignal) {
```
```suggestion
  public void receiveGpsSignals(Satellite satelliteSignal); {
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsManager.java:
```diff
@@ -0,0 +1,12 @@
+
+public interface DisplayGpsManager {
+
+  public String receiveGpsSignalStrengths(Satellite satelliteSignal);
```
```suggestion
  public String receiveGpsSignalStrength(Satellite satelliteSignal);
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java:
```diff
@@ -0,0 +1,17 @@
+
+public class DisplayGpsManagerImplementation implements DisplayGpsManager{
+    
+  public String receiveGpsSignalStrengths(Satellite satelliteSignal) {
```
```suggestion
  public String receiveGpsSignalStrength(Satellite satelliteSignal) {
```

andrewdimmer reviewed (changes requested):
Couple of spelling changes, couple of package issues, and a small  breaking change to the UML model based on a typo before we are ready to merge this one.

andrewdimmer committed:
[Display_01] Fixed Typos in Public Interface Documentation

andrewdimmer committed:
Generate SVG and PNG Images for PlantUML Diagrams

atpoynter committed:
Update src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

atpoynter committed:
Update src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

reedsikorskiOU committed:
Final Changes

Andrew Suggested Final Typos, fixed now

reedsikorskiOU committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

reedsikorskiOU committed:
Merge branch 'main' into display01-public-interfaces

atpoynter committed:
[Display01] Corrected typos and class mispellings for DisplayGpsManagerImplementation.java and DisplayGpsManager.java

atpoynter committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

EC01EC committed:
Merge branch 'main' into display01-public-interfaces

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java:
```diff
@@ -0,0 +1,15 @@
+
+public interface DisplayGpsInterface {
+
+  public void recieveGpsSignals(Satellite satelliteSignal);
```
```suggestion
  public void receiveGpsSignal(Satellite satelliteSignal);
```

andrewdimmer commented on docs/uml-class-diagrams/display01/production/DisplayGpsInterface/DisplayGpsInterface.puml:
```diff
@@ -4,7 +4,7 @@ package edu.oakland.production.display01 {
         __
         .. Use Case 1 ..
-        {abstract} + void revieveGPSSignals(Satellite satelliteSignal)
+        {abstract} + void recieveGpsSignal(Satellite satelliteSignal)
```
```suggestion
        {abstract} + void receiveGpsSignal(Satellite satelliteSignal)
```

andrewdimmer commented on src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java:
```diff
@@ -0,0 +1,23 @@
+
+public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+
+  public void recieveGpsSignals(Satellite satelliteSignal) {
```
```suggestion
  public void receiveGpsSignal(Satellite satelliteSignal) {
```

andrewdimmer reviewed (changes requested):
Few more typos, should be good to merge once these are fixed though!

EC01EC committed:
[Display_01] Created DisplayGpsReceiver

Added DisplayGpsReceiver and DisplayGpsReceiverImplementation

EC01EC committed:
Apply suggestions from code review

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

EC01EC committed:
Generate SVG and PNG Images for PlantUML Diagrams

EC01EC committed:
Fixed Style Guide on DisplayGpsReceiver

Fixed indentation to match style guide on DisplayGpsReceiver and DisplayGpsReceiverImplementation

EC01EC committed:
Merge branch 'display01-public-interfaces' of https://github.com/andrewdimmer/csi-3370-software-project into display01-public-interfaces

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Added Javadoc comments to DisplayGpsReceiverImplementation

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Fixed Javadoc comments on DisplayGpsReceiverImplementation

EC01EC committed:
Updated SatelliteSignalCheckRequest and DisplayGpsReceiverImplementation

Added "public" modifier to getCheckType method in SatelliteSignalCheckRequest.java

Fixed Javadoc errors in DisplayGpsReceiverImplementation.java

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

"Fixed" Javadoc errors in DisplayGpsReceiverImplementation

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

"fixes"

EC01EC committed:
Update DisplayGpsReceiverImplementation.java

Added "implements DisplayGpsReceiver" to class line

andrewdimmer committed:
[Display_01] Finally Fixed JavaDoc Checkstyle

andrewdimmer committed:
Update DisplayGpsReceiverImplementation.java

andrewdimmer reviewed (approved):
Looks good to me, you're all set to merge!

EC01EC committed:
Merge branch 'main' into display01-public-interfaces

EC01EC reviewed (approved):
looks good
